mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSNode.kt

```diff
@@ -220,8 +220,7 @@ data class MPSNode(val node: SNode) : IDefaultNodeAdapter, IReplaceableNode {
                 ?: SReferenceLinkAdapterById(SReferenceLinkId.deserialize(link.getUID()), "")
         }
 
-        val targetNode = target?.let { getArea().resolveNode(it.reference) } as MPSNode
-        node.setReferenceTarget(refLink, targetNode.node)
+        node.setReferenceTarget(refLink, (target as? MPSNode?)?.node)
```

Comment:
Why did this change happen?

Reply:
It's an existing bug, I stumbled upon during implementation. @odzhychko also found it. See #728

Reply:
You can't just set the reference to null if a not null target was provided. This case has to result in an exception.

Reply:
You are right, the safe-cast operator shouldn't be there. 
I'll make it even more explicit:
```kotlin
require(target is MPSNode?) { "target has to be an MPSNode or null" }
val refLink = // ...
node.setReferenceTarget(refLink, target?.node)
```

Reply:
@slisson fine for you this way?

Reply:
I'll drop this commit. As it was included in #728, which also adds test cases for this.